File: Server_trunk/res/scripts/cell/Spell/Effects/EffectDynamicDamage.py
# -*- coding: utf-8 -*-
#
"""
"""
from KBEDebug import *
import GloballyDefine as GD

from ..Spell import SpellEffect
from Extra.FightSystem import FightSystem

class EffectDynamicDamage(SpellEffect):
	"""
	伤害效果
	"""
	def init(self, dataSection):
		"""
		"""
		SpellEffect.init(self, dataSection)

		# 伤害类型（物理、法术）
		self.damageType = dataSection.readInt( "param1" )
		
		# 伤害百分比(float)
		self.damagePercent = dataSection.readFloat( "param2" )
		
		# 伤害修正值(int)
		self.damageValue = dataSection.readInt( "param3" )
		
	def cast(self, src, dst, spell):
		"""
		src向dst施放一个效果
		@param src: Entity; 施法者
		@param dst: Entity; 受术者
		@param spell: instance of Spell; 表示是由哪个技能触发的效果
		@return: None
		"""
		FightSystem.fight( src, dst, spell, self.damageType, self.damagePercent, self.damageValue )

File: Client_trunk2/Assets/Scripts/Spells/Effects/EffectDynamicDamage.cs
using UnityEngine;
using System.Collections;

namespace SPELL
{
    /// <summary>
    /// 伤害效果
    /// </summary>
    public class EffectDynamicDamage : SpellEffect
    {
        public int damageValue;
        public float damageVol;
        public float dropPercent;
        public int magicSparCount;
        public CDeadType deadType;

        public override void Init(DataSection.DataSection dataSection)
        {
            base.Init(dataSection);
            damageValue = dataSection.readInt("param1");
            damageVol = dataSection.readFloat("param2");
            dropPercent = dataSection.readFloat("param3");
            magicSparCount = dataSection.readInt("param4");
            deadType = (CDeadType)dataSection.readInt("param5");
        }

        public override void Cast(AvatarComponent src, AvatarComponent dst, SpellEx spell, SpellTargetData targetData)
        {
            //概率获得魔法晶石
            if (magicSparCount > 0 && Random.value < dropPercent)
            {
                MagicSparMgr.CreateMagicSpar(dst, magicSparCount);
            }

            FightSystem.Fight(src,dst, spell, damageValue, damageVol, deadType);
        }
    }
}
